controllers/compaction/reconciler.go

```diff
@@ -298,15 +299,39 @@ func (r *Reconciler) createCompactionJob(ctx context.Context, logger logr.Logger
 						Image:           *etcdBackupImage,
 						ImagePullPolicy: v1.PullIfNotPresent,
 						Args:            getCompactionJobArgs(etcd, r.config.MetricsScrapeWaitDuration.String()),
-						VolumeMounts:    getCompactionJobVolumeMounts(etcd, logger),
-						Env:             getCompactionJobEnvVar(etcd, logger),
 					}},
-					Volumes: getCompactionJobVolumes(ctx, r.Client, logger, etcd),
 				},
 			},
 		},
 	}
 
+	if vms, err := getCompactionJobVolumeMounts(etcd); err != nil {
+		return nil, fmt.Errorf("error while creating compaction job in %v for %v : %v",
+			etcd.Namespace,
+			etcd.Name,
+			err)
+	} else {
```

Comment:
`else` is not necessary since you are returning in the `if` block, but I understand that the scope of the condition evaluation, ie, `vms` variable gets carried over to the else block as well. Usually in any of the golang code in gardener/kubernetes, it's done like this. Not a compulsion, but you can consider it:
```
            vms, err := getCompactionJobVolumeMounts(etcd)
            if err != nil {
                return nil, fmt.Errorf("error while creating compaction job in %v for %v : %v",
			etcd.Namespace,
			etcd.Name,
			err)
	    }
            job.Spec.Template.Spec.Containers[0].VolumeMounts = vms
```

Reply:
Two reasons:
1. I got to use the nice feature of go lang `if vms, err := getCompactionJobVolumeMounts(etcd); err != nil {` to put assignment and condition in same line. It always looks more appealing and cleaner to me
2. As the scope is only limited to the if condition, garbage collector does not need to preserve the scope of the variable `vms` till the end of the function but only within the if block. So, an extra variable is free just after the `if` block.